src/Adapter/Presenter/Product/ProductLazyArray.php

```diff
@@ -574,18 +595,59 @@ private function shouldShowPrice(
      *
      * @param array $product
      *
-     * @return mixed
+     * @return bool
      */
-    private function shouldShowAddToCartButton($product)
+    private function shouldShowAddToCartButton($product): bool
     {
         return (bool) $product['available_for_order'];
     }
 
-    private function fillImages(
-        ProductPresentationSettings $settings,
-        array $product,
-        Language $language
-    ) {
+    /**
+     * @param array $product
+     *
+     * @return bool
+     */
+    private function shouldShowOutOfStockLabel(array $product): bool
+    {
+        if (!(bool) $this->configuration->get('PS_SHOW_LABEL_OOS_LISTING_PAGES')) {
+            return false;
+        }
+        // Displayed only if the order of out of stock product is denied.
```

Comment:
Usually all this is handled via `Product::isAvailableWhenOutOfStock($product['out_of_stock'])`